### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by TarasFirst**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,13 +1,21 @@
...
+            "name": error["code"],
+            "source": "flake8"
+            }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+
+    return {"errors": [format_linter_error(i) for i in errors],
```
Comment (ArsenPidhoretskyi) [inline app/main.py]: rename i to error

File: app/main.py [hunk 2]
```diff
@@ -1,13 +1,21 @@
...
+            "path": file_path,
+            "status": ("failed" if errors else "passed")}
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+
+    return [format_single_linter_file(i, linter_report[i])
+            for i in linter_report.keys()]
```
Comment (ArsenPidhoretskyi) [inline app/main.py]: Use .items() and rename i

**Review by ArsenPidhoretskyi — CHANGES_REQUESTED** [review]

File: app/main.py (hunk 2 above)
Comment (TarasFirst) [reply app/main.py:21]: Is it necessary to use .items(), if i need only keys ?

File: app/main.py [hunk 3]
```diff
@@ -1,13 +1,21 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+
+    return {"line": error["line_number"],
```
Comment (Polyakiv-Andrey) [inline app/main.py]: After opening the parentheses, write all the arguments on a new line, including the first one

File: app/main.py [hunk 4]
```diff
@@ -1,13 +1,21 @@
...
+            "name": error["code"],
+            "source": "flake8"
+            }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+
+    return {"errors": [format_linter_error(error) for error in errors],
```
Comment (Polyakiv-Andrey) [inline app/main.py]: After opening the parentheses, write all the arguments on a new line, including the first one, the closing bracket is also written on a new line

File: app/main.py [hunk 5]
```diff
@@ -1,13 +1,21 @@
...
+            "path": file_path,
+            "status": ("failed" if errors else "passed")}
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+
+    return [format_single_linter_file(file_path, linter_report[file_path])
```
Comment (Polyakiv-Andrey) [inline app/main.py]: After opening the parentheses, write all the arguments on a new line, including the first one, the closing bracket is also written on a new line

**Review by Polyakiv-Andrey — CHANGES_REQUESTED** [review]

**Comment by TarasFirst** [comment]
Should i debug something else?

**Pull request closed without merging**